src/wazuh_modules/syscollector/src/syscollectorImp.cpp

```diff
@@ -1709,6 +2015,23 @@ void Syscollector::persistDifference(const std::string& id, Operation operation,
     }
     else if (m_spSyncProtocol)
     {
+        m_spSyncProtocolVD->persistDifference(id, operation, index, data, version);
```

Comment:
The conditional (https://github.com/wazuh/wazuh/pull/33285/files#diff-499a711b9b6151617ff3215925f8e57f321a0b51dbb0b13d51dc5465cf9689d9R2016) should check m_spSyncProtocolVD instead of m_spSyncProtocol